app/routes/api/users.js

```diff
@@ -108,4 +108,37 @@ router.post('/:username', function(req, res, next){
   });
 });
 
+//Update users password
```

Comment:
you just copy pasted this route, but the one above it also matches /:username ....

you should just add the edit password stuff to the existing edit user route